inhibit/inhibit.go

```diff
@@ -98,17 +93,41 @@ func (ih *Inhibitor) Run() {
 	}
 }
 
+// Run the Inihibitor's background processing.
+func (ih *Inhibitor) Run() {
+	var (
+		g   group.Group
+		ctx context.Context
+	)
+
+	ctx, ih.cancel = context.WithCancel(context.Background())
+	gcCtx, gcCancel := context.WithCancel(ctx)
+	runCtx, runCancel := context.WithCancel(ctx)
+
+	g.Add(func() error {
+		ih.runGC(gcCtx)
+		return nil
```

Comment:
If you return the context error, you can return it here. And the same for the ih.run below.